batch/test/test_utils.py

```diff
@@ -35,3 +36,18 @@ def test_gcp_worker_memory_per_core_mib():
     assert gcp_worker_memory_per_core_mib('n1', 'highmem') == 6656
     assert gcp_worker_memory_per_core_mib('n1', 'highcpu') == 924
     assert gcp_worker_memory_per_core_mib('g2', 'standard') == 4000
+
+
+def test_gcp_resource_from_dict():
+    name = 'accelerator/l4-nonpreemptible/us-central1/1712657549063'
+    gpu_data_dic_single = {'name': name, 'number': 1, 'type': 'gcp_acceleratorV2'}
+    resource = gcp_resource_from_dict(gpu_data_dic_single)
+    quantified_resources = resource.to_quantified_resource(1000, 20, 1024, 20)
+    assert quantified_resources
+    assert quantified_resources['quantity'] == 1024
+
+    gpu_data_dic_double = {'name': name, 'number': 2, 'type': 'gcp_acceleratorV2'}
+    resource = gcp_resource_from_dict(gpu_data_dic_double)
+    quantified_resources = resource.to_quantified_resource(1000, 20, 1024, 20)
+    assert quantified_resources
+    assert quantified_resources['quantity'] == 2048
```

Comment:
This test is great. Can you also add another test that verifies that the different g2 machine types have the correct number of GPUs? (Can just test `machine_type_to_gpu_num‘